lib/web_of_science/links_client.rb

```diff
@@ -0,0 +1,85 @@
+module WebOfScience
+  # Links AMR (Article Match Retrieval) Service
+  # @see http://ipscience-help.thomsonreuters.com/LAMRService/WebServicesOverviewGroup/overview.html Service documentation
+  class LinksClient
+
+    LINKS_HOST = 'https://ws.isiknowledge.com'.freeze
+    LINKS_PATH = '/cps/xrpc'.freeze
+
+    attr_reader :username, :password, :host
+
+    # @param [String] username
+    # @param [String] password
+    # @param [String] host
+    def initialize(username: nil, password: nil, host: LINKS_HOST)
+      @host = host
+      @username = username
+      @password = password
+      (@username, @password) = Base64.decode64(Settings.WOS.AUTH_CODE).split(':', 2) unless username
+    end
+
+    # Retrieve identifier 'fields' for the record 'ids'
+    # @param [Array<String>] ids
+    # @param [Array<String>] fields (defaults to ['ut', 'doi', 'pmid'])
+    # @return [Hash<String => Hash>]
+    def links(ids, fields = %w(ut doi pmid))
+      ids.uniq!
+      raise ArgumentError, '1-50 ids required' if ids.empty? || ids.count > 50
```

Comment:
If you care about `ids` being `nil`, you could consider changing `#empty?` to `#blank?` though I don't know if this is likely to be problematic in this codebase.

Reply:
If you want to handle all of the problematic cases (not passed an array, passed an empty array, passed an array with > 50 elements, you could change this to:

```ruby
# ...
raise ArgumentError, '1-50 ids required' unless valid_identifier_list?(ids)
# ...

def valid_identifier_list?(ids)
  return false unless ids.is_a? Enumerable
  ids.count.between?(1, 50)
end
```

Reply:
OK, open to that.  Also note that it runs an `id.uniq` on them so it makes sense to get the uniq set before counting them.

Reply:
OK, done.